cmake/FindCLN.cmake

```diff
@@ -67,7 +67,7 @@ if(NOT CLN_FOUND_SYSTEM)
     BUILD_BYPRODUCTS <INSTALL_DIR>/lib/libcln.a
   )
 
-  add_dependencies(CLN-EP GMP)
+  add_dependencies(CLN-EP GMP_SHARED)
```

Comment:
Are GMP_SHARED and GMP_STATIC always built? How about moving add_dependencies down to the `if(ENABLE_STATIC_LIBRARY)` block and either add GMP_STATIC or GMP_SHARED as dependency?